### DISCUSSION TITLE
Infinity as the min-y parameter in viewBox????

### DISCUSSION THREAD

**Discussion by stuvjordan-uroc** [d3/d3#3943] [Q&A] [answered]
I'm trying to understand how [this d3 tree code](https://observablehq.com/@d3/tree-component#Tree) works, step-by-step.

There's a step that has me stuck, and I'm hoping someone can explain it.

So, first of all, the code first sets variables dx and x0 , like this:
```
const dx = 10;
let x0 = Infinity;
```
It then creates an svg with a viewBox attribute set like this:
```
const svg = d3.create("svg")
      .attr("viewBox", [-dy * padding / 2, x0 - dx, width, height])
```
The variables `dy`, `padding`, `width` and `height` are set elsewhere in code that I'm omitting.  What I'm hoping someone can help me understand is the part where the min-y parameter is being set to `x0-dx`.

I think (please correct me if this is wrong) that `x0-dx` will evaluate to `Infinity`. 

My understanding is that min-y is the smallest y coordinate in the svg.  And I'm used to setting it to 0, so that if I assign something a positive y coordinate, it will show up somewhere below the top of the viewBox.

But what happens if I set min-y to Infinity????  How do I interpret y coordinates in that case?

Is this some special hack that d3 people like to do?  If so, where is it documented or explained?  I don't see anything in [MDN's article on viewBox ](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/viewBox) about the consequences of setting min-y to Infinity.

Thanks!

**Comment by mbostock** [comment, marked as answer]
The values of `x0` and `x1` are computed by iterating over the tree; their value is no longer infinity after iterating; instead they are the minimum and maximum node x-value. Initializing `x0` and `x1` to ±Infinity simplifies the computation of the minimum and maximum values. It’s the same reason `Math.min()` (with no arguments) returns Infinity and likewise `Math.max()` returns -Infinity.

```js
  // Center the tree.
  let x0 = Infinity;
  let x1 = -x0;
  root.each(d => {
    if (d.x > x1) x1 = d.x;
    if (d.x < x0) x0 = d.x;
  });
```

**Reply by stuvjordan-uroc** [reply]
argh...thanks.  Dumb question on my part.  I missed that x1 and x0 were being overwritten in that centering step!
